server/api/models.py

```diff
@@ -51,7 +55,78 @@ class Spell(models.Model):
     archetype = models.TextField()
     circles = models.TextField()
 
+class Archetype(models.Model):
+    slug = models.SlugField(unique=True, default=uuid.uuid1)
+    name = models.TextField(primary_key=True)
+    desc = models.TextField()
+
+class CharClass(models.Model):
+    slug = models.SlugField(unique=True, default=uuid.uuid1, primary_key=True)
+    name = models.TextField()
+    hit_dice = models.TextField()
+    hp_at_1st_level = models.TextField()
+    hp_at_higher_levels = models.TextField()
+    prof_armor = models.TextField()
+    prof_weapons = models.TextField()
+    prof_tools = models.TextField()
+    prof_saving_throws = models.TextField()
+    prof_skills = models.TextField()
+    equipment = models.TextField()
+    table = models.TextField()
+    desc = models.TextField()
+    spellcasting_ability = models.TextField()
+    subtypes_name = models.TextField()
+    archetype = models.ForeignKey(Archetype, on_delete=models.CASCADE, null=True)
+
+class SubRace(models.Model):
+    slug = models.SlugField(unique=True, default=uuid.uuid1, primary_key=True)
+    name = models.TextField()
+    desc = models.TextField()
+    asi_desc = models.TextField()
+    asi = JSONField()
+    traits = models.TextField()
+
+class Race(models.Model):
+    slug = models.SlugField(unique=True, default=uuid.uuid1, primary_key=True)
+    name = models.TextField()
+    desc = models.TextField()
+    asi_desc = models.TextField()
+    asi = JSONField()
+    age = models.TextField()
+    alignment = models.TextField()
+    size = models.TextField()
+    speed = JSONField()
+    speed_desc = models.TextField()
+    languages = models.TextField()
+    vision = models.TextField()
+    traits = models.TextField()
+    subtypes  = models.TextField()
+    sub_race = models.ForeignKey(SubRace, on_delete=models.CASCADE, null=True)
+
+class Plane(models.Model):
+    slug = models.SlugField(unique=True, default=uuid.uuid1)
```

Comment:
It might make sense to do some refactoring in the models, since basically everything is using a slug, a name, and a desc. you can move that all into a parent class that these classes inherit from and save a few dozen repeated lines :)

Reply:
As a note to this, perhaps we should agree on a few universal fields to add to the parent class?  Some metadata as well (like my document source)?
1. Name - (E.G. "Barbarian" or "Blinded")
2. Slug - ("magic-missile")  Consistent, for the URL.
3. Description - The Body of the document.  Lots of differences here, but here's a question:
Should the Description contain the Name at the top?  I was inconsistent here,  For things like Subtype I did not, for things like Conditions I did.
4. Meta_Source Work (this is a reference back to some SRD object sitting in a table somewhere.
Here's some questionable shared fields:
5. Meta_Page  (This would be the PHB, DMG, or MM page.)
6. Meta_Section or Meta_Parent or something... this is a way to organize the pages into navigation, although some of the sections kind of defy organization.

Reply:
I like these! Along with the obligatory `last_edited` and `date_created` fields of course. And as for the description - I'd recommend not including the name - that way we can compute a description block that includes the name if we want to, but either way we won't have it listed twice.

Reply:
Yeah, agree on leaving the name out of the description. Generally speaking you'd want to be able to populate that in a tooltip or other display, probably referenced by the name.

Agree on all those universal fields